### Using the function arange, create an array of 10 integers 'd' from -20 to 25 (inclusive), in steps of 5.

In [1]:
import numpy as np
d = np.arange(start = -20, stop = 26, step =5)
print(d)

[-20 -15 -10  -5   0   5  10  15  20  25]


### Multiply all values in d by 10, storing the result in d2.

In [2]:
d2 = d*10
print(d2)

[-200 -150 -100  -50    0   50  100  150  200  250]


### Add the values stored in d and d2, element-wise, printing the result to the console

In [3]:
print(d+d2)

[-220 -165 -110  -55    0   55  110  165  220  275]


### Reshape d2 into a 5x2 array, storing the result in d2.

In [4]:
d2 = d2.reshape(5,2)
print(d2)

[[-200 -150]
 [-100  -50]
 [   0   50]
 [ 100  150]
 [ 200  250]]


### Create an array d3, containing the integers 1,2,3,4,5.

In [5]:
d3 = np.arange(1,6)
print(d3)


[1 2 3 4 5]


### Reshape d3 into the necessary shape such that d2*d3 succeeds (hint: broadcasting). Save the result of d2*d3 into d4, and print d4 to the console.

In [6]:
d3 = d3.reshape(5,1)

d4 = d2*d3

print(d4)


[[-200 -150]
 [-200 -100]
 [   0  150]
 [ 400  600]
 [1000 1250]]


### Print to the console the 1st, 3rd and 5th rows, both columns, of d4.

In [7]:
print(d4[[0,2,4],:])

[[-200 -150]
 [   0  150]
 [1000 1250]]


### Print the object ID of d4 to the console.

In [8]:
print(f"ID of d4 :{id(d4)}")

ID of d4 :2228165918960


### Resize (not reshape) d4 into a 2x5 array.

In [9]:
d4.resize(2,5)
print(d4)

[[-200 -150 -200 -100    0]
 [ 150  400  600 1000 1250]]


### Flatten (not reshape) d4 back into a 10x1 array. Print the result to the console. Your output should match: [-200 -150 -200 -100 0 150 400 600 1000 1250]

In [10]:
d4 = d4.flatten()
print(d4)

[-200 -150 -200 -100    0  150  400  600 1000 1250]
